Tests/ResearchRegressionTests.cs

```diff
@@ -0,0 +1,287 @@
+                        if (line != null && line.StartsWith("    }"))
+                        {
+                            lines.Add(line);
+                            break;
+                        }
+                    }
+                }
+                else
+                {
+                    lines.Add(line);
+                }
+            }
+
+            file.DisposeSafely();
+            File.WriteAllLines(templatePath, lines);
+        }
+
+        /// <summary>
+        /// Split long string into multiple lines
+        /// </summary>
+        /// <remarks>Doesn't break a line if it ends with "\"</remarks>
+        private static List<string> SplitIntoMultipleLines(string longInput, int maxChunkSize)
+        {
+            List<string> resultLines = new();
+            for (int i = 0; i < longInput.Length;)
+            {
+                var chunk = longInput.Substring(i, Math.Min(maxChunkSize, longInput.Length - i));
+                i += maxChunkSize;
+                while (chunk.EndsWith("\\") && i < longInput.Length)
+                {
+                    chunk += longInput[i];
+                    i++;
+                }
+                resultLines.Add($"            \"{chunk}\" +");
+            }
+            // use ; for endline
+            var lastLine = resultLines.Last();
+            resultLines[resultLines.Count - 1] = lastLine.Remove(lastLine.Length - 2) + ";";
+            return resultLines;
+        }
+
+        private static string CleanDispensableEscapeCharacters(string json)
+        {
+            json = json
+                .Replace("\\n", string.Empty)
+                .Replace("\n", string.Empty)
+                .Replace("\\r", string.Empty)
+                .Replace("\r", string.Empty)
+                .Replace("\\t", string.Empty)
+                .Replace("\t", string.Empty);
+            return json;
+        }
+
+        private static string RunResearchNotebookAndGetOutput(string notebookPath, string notebookoutputPath, string workingDirectoryForNotebook)
+        {
+            var args = $"-m papermill \"{notebookPath}\" \"{notebookoutputPath}\" --log-output --cwd {workingDirectoryForNotebook}";
+
+            // Use ProcessStartInfo class
+            var startInfo = new ProcessStartInfo(_pythonLocation, args)
+            {
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                UseShellExecute = false,
+                WindowStyle = ProcessWindowStyle.Hidden,
+                CreateNoWindow = true,
+                WorkingDirectory = Directory.GetCurrentDirectory()
+            };
+
+            var process = new Process
+            {
+                StartInfo = startInfo,
+            };
+            process.Start();
+            process.BeginErrorReadLine();
+            process.BeginOutputReadLine();
+            process.WaitForExit();
+            Assert.AreEqual(0, process.ExitCode);
```

Comment:
Ah suggest let's move this check after checking for the output else it will fail here and not really provide much information for the user